### PR TITLE
Fix global state for authentication

### LINKED ISSUE #322: Bug: incomplete logout when using several tabs
**Describe the bug**
When Argus is running in several tabs in the browser, logout in one of them does not trigger the logged out state in the rest of the tabs. 
API calls can be performed in all tabs, where logout was not performed explicitly. 

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to /incidents
2. Open /timeslots in a new tab
3. Perform logout in the /incidents tab
4. Perform any update on a timeslot in the /timeslots tab
5. See the update in the /timeslots tab being successfully performed

**Expected behavior**
Logout from Argus should trigger logout in all tabs running Argus with a given cookie.

### PR DESCRIPTION
**Pull request by podliashanyk**
Changes made:

- [x] Call logout endpoint
- [x] Use token that is stored in browser cookies
- [x] Update redirects
- [x] Update tests

Closes #322, #79, #175

### CODE REVIEW

**Comment by podliashanyk** [comment]
Implemented changes:

- Logout endpoint is called
- Token from browser cookies is used in order to check for whether session is valid
- Protected route redirect (to login page) is fixed
- Implemented redirect to /incidents from login is user is already authenticated

As a result:

- login/logout is global, regardless how many Argus tabs/windows (of the same browser) user has opened. Login/logout in one of the tabs is correctly reflected in all active Argus tabs/windows (of the same browser)
- redirects correctly reflect authentication state

File: src/store.tsx [hunk 1]
```diff
@@ -1,35 +0,0 @@
...
-    default:
-      return initialState;
-  }
-}
-
-export function StoreProvider(props: any) {
-  const [state, dispatch] = React.useReducer(reducer, initialState);
-  const values = { state, dispatch };
-  return <Store.Provider value={values}>{props.children}</Store.Provider>;
-}
```
Comment (hmpf) [inline src/store.tsx:35]: Getting rid of this was a good choice considering the naming: "store", but only of auth-relevant things.

**Review by hmpf — COMMENTED** [review]
Seems to be working as advertised.

**Review by hmpf — APPROVED** [review]
Ugh, hit "Submit" to soon.

**Pull request merged**